File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cpp
#include <string>
#include <vector>

using namespace std;

int transI(string time){
    string m = time.substr(0,2);
    string s = time.substr(3);
    return stoi(m)*60 + stoi(s);
}

string transS(int time){
    string result = "";
    int s = time%60;
    int m = time/60;
    if(m < 10) result+="0";
    result+=(to_string(m) + ":");
    if(s < 10) result+="0";
    result+=(to_string(s));
    return result;
}

string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {
    int videoLenI = transI(video_len);
    int posI = transI(pos);
    int opStartI = transI(op_start);
    int opEndI = transI(op_end);
    if(posI >= opStartI && posI <= opEndI) posI = opEndI;
    for(auto s : commands){
        if(s == "next"){
            posI+=10;
            if(posI > videoLenI) posI = videoLenI;
        }
        else if(s == "prev"){
            posI-=10;
            if(posI < 0) posI = 0;
        }
        if(posI >= opStartI && posI <= opEndI) posI = opEndI;
    }
    return transS(posI);
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        String answer = "";
        int c = toSec(pos);
        int last = toSec(video_len);
        int ops = toSec(op_start);
        int ope = toSec(op_end);
        for(String command : commands){
            if(c >= ops && c < ope) c = ope;
            if(command.equals("prev")){
                c-=10;
                if(c<0) c = 0;
            }else if(command.equals("next")){
                c+=10;
                if(c > last) c = last;
            }
        }
        if(c >= ops && c < ope) c = ope;
        return toMin(c);
    }
    
    public static String toMin(int sec){
        StringBuilder sb = new StringBuilder();
        if(sec/60 < 10) sb.append("0");
        sb.append(sec/60).append(":");
        if(sec%60 < 10) sb.append("0");
        sb.append(sec%60);
        return sb.toString();
    }
    
    public static int toSec(String min){
        int m = Integer.parseInt(min.substring(0,2));
        int s = Integer.parseInt(min.substring(3,5));
        return m*60+s;
    }
}